lib/dal/src/action.rs

```diff
@@ -184,6 +232,31 @@ impl Action {
         Ok(result)
     }
 
+    pub async fn prototype_id(
+        ctx: &DalContext,
+        action_id: ActionId,
+    ) -> ActionResult<Option<ActionPrototypeId>> {
```

Comment:
Why an `Option`? If there's an `Action` without an `ActionPrototype` that seems like things are very broken, and not really recoverable, or we're in the middle of creating the `Action` where we should already know which `ActionPrototype` we'd be setting things up with.

Reply:
That makes sense will change.

Btw why is component optional for the action?

Reply:
Possibly premature future planning: I can envision Actions that are scoped to the entire workspace (having `ActionPrototype` linked to the workspace itself, in addition to the ones linked to the `SchemaVariant`s).